### PR TITLE
Adding prints for better dubugging and adding comments. Making output writing independent of link extraction.

### PR DESCRIPTION
**Pull request by deyarchit**
The the general exceptions previously added could cause some confusions while debugging. Hence adding Exception name along with the message.

Added description for process url method. 
Proposal: Can we bring the the output writing statement outside the if condition and remove the url key from the dictionary.

### CODE REVIEW

**Comment by rohan-achar** [comment]
That proposal is great! Submit patch with that please so that it can be attributed to you.

File: Crawler4py/Fetcher.py [hunk 1]
```diff
@@ -55,14 +55,21 @@ def FetchUrl(self, url, depth, urlManager, retry = 0):
...
 
     
     def __ProcessUrlData(self, url, htmlData, depth, urlManager):
+        '''
+        Function to extract information from the URL. 
+        The text data, html data and url are added to the output buffer. The url's found on the page are sent to the Frontier addition. 
+        '''
         textData = self.config.GetTextData(htmlData, forUrl=url)
+        # Sending the relevant information to the output writing function
+        urlManager.AddOutput({"html": htmlData, "text": textData, "url": url})
```
Comment (deyarchit) [inline Crawler4py/Fetcher.py:69]: Contrary to what I initially said, we can send all the information, htmlData, textData and the url even if we move this call outside the link extraction.

**Comment by deyarchit** [comment]
Added the proposal along with the last commit.

Also, I feel since you have put all the prior relevant changes in the CHANGELOG, the key commits that have happened during this quarter should also be added to it.

**Comment by rohan-achar** [comment]
will add the CHANGELOG changes in summary after this :)

**Comment by rohan-achar** [comment]
Thanks for the patch.

**Pull request merged**